packages/create-toolpad-app/src/templates/auth/auth.ts

```diff
@@ -18,7 +18,33 @@ const CredentialsProviderTemplate = `Credentials({
   },
 }),`;
 
-const oAuthProviderTemplate = (provider: SupportedAuthProvider) => `${provider}({
+const checkEnvironmentVariables = (providers: SupportedAuthProvider[]) => `
+const missingVars: string[] = [];
+${providers
+  .filter((p) => p !== 'credentials')
+  .map(
+    (provider) => `
+if (!process.env.${kebabToConstant(provider)}_CLIENT_ID) {
+  missingVars.push('${kebabToConstant(provider)}_CLIENT_ID');
+}
+if (!process.env.${kebabToConstant(provider)}_CLIENT_SECRET) {
+  missingVars.push('${kebabToConstant(provider)}_CLIENT_SECRET');
+}`,
+  )
+  .join('\n')}
+
+if (missingVars.length > 0) {
+  const baseMessage = 'Authentication is configured but the following environment variables are missing:';
+  
+  if (process.env.NODE_ENV === 'production') {
```

Comment:
Avoid different behavior in dev vs. prod. If it crashes in prod it must crash in dev. Let's just warn then.


I'd suggest to start by just generating the following code
```tsx
function warnOnMissing(var: string, provider: string, name: string) {
  if (!var) {
    console.warn(`${provider} provider: environment variable "${name}" missing`);
  }
}

warnOnMissing(process.env.GOOGLE_CLIENT_ID, 'Google', 'GOOGLE_CLIENT_ID')
warnOnMissing(process.env.GOOGLE_CLIENT_SECRET, 'Google', 'GOOGLE_CLIENT_SECRET')
// ...
```